packages/manager/apps/catalog-revamp/src/components/Filters/Filters.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { useState, useRef, useEffect } from 'react';
+import {
+  OsdsButton,
+  OsdsText,
+  OsdsIcon,
+  OsdsCheckbox,
+  OsdsCheckboxButton,
+  OsdsLink,
+  OsdsSkeleton,
+} from '@ovhcloud/ods-stencil/components/react/';
+import {
+  OdsIconName,
+  OdsIconSize,
+  OdsButtonVariant,
+  OdsButtonType,
+  OdsTextLevel,
+  OdsButtonSize,
+  OdsCheckboxButtonSize,
+  OdsTextSize,
+} from '@ovhcloud/ods-core';
+import { OdsThemeColorIntent } from '@ovhcloud/ods-theming';
+import { useTranslation } from 'react-i18next';
+import { useCatalog } from '@/hooks/useCatalog';
+import { getAvailableCategories, getUniverses } from '@/utils/utils';
+import './Filters.scss';
+
+const FilterItem: React.FC<{ label?: string; loading?: boolean }> = ({
+  label,
+  loading,
+}) => {
+  return (
+    <div className="filter-item">
+      <OsdsCheckbox>
+        <OsdsCheckboxButton
+          size={OdsCheckboxButtonSize.sm}
+          color={OdsThemeColorIntent.primary}
+        >
+          <span slot="end">{loading ? <OsdsSkeleton /> : label}</span>
+        </OsdsCheckboxButton>
+      </OsdsCheckbox>
+    </div>
+  );
+};
+
+const Filters: React.FC = () => {
+  const [categories, setCategories] = useState<string[]>([]);
+  const [universes, setUniverses] = useState<string[]>([]);
+  const [searchText, setSearchText] = useState<string>('');
+  const { products } = useCatalog({
+    categories,
+    universes,
+    searchText,
+  });
+  const reset = () => {
+    setCategories([]);
+    setUniverses([]);
+    setSearchText('');
+  };
+
+  useEffect(() => {
+    setCategories(getAvailableCategories(products, universes));
+    setUniverses(getUniverses(products));
+  }, [universes, searchText, products, categories]);
+
+  const [showFilters, setShowFilters] = useState(true);
```

Comment:
**nitpick:** Can be moved to top with all the `useState` declarations to have better readability

Reply:
True, done